### PR TITLE
feat: support for span evaluations to get evaluations endpoint

### LINKED ISSUE #2894: [persistence] wire evaluations GET endpoint to db

### PR DESCRIPTION
**Pull request by axiomofjoy**
- adds support for span annotations to the GET evaluations endpoint.
- fixes bug with query name parameter for `get_evaluations` client function

resolves #2894

### CODE REVIEW

File: src/phoenix/server/api/routers/v1/evaluations.py [hunk 1]
```diff
@@ -105,25 +116,71 @@ async def get_evaluations(request: Request) -> Response:
...
-    project = traces.get_project(project_name)
-    if not project:
+
+    db: Callable[[], AsyncContextManager[AsyncSession]] = request.app.state.db
+    async with db() as session:
+        span_evaluation_names = await session.scalars(
+            select(distinct(models.SpanAnnotation.name)).where(
+                models.SpanAnnotation.annotator_kind == "LLM"
+            )
+        )
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py:132]: Good call.

[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py:132]: What would the default be?

[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py]: addressed in cc2ebd9c4d85398ecbd255a49cbf3cb4950c713c

File: src/phoenix/server/api/routers/v1/evaluations.py [hunk 2]
```diff
@@ -105,25 +116,71 @@ async def get_evaluations(request: Request) -> Response:
...
+            .where(
+                and_(
+                    models.Project.name == project_name,
+                    models.SpanAnnotation.name == evaluation_name,
+                    models.SpanAnnotation.annotator_kind == "LLM",
+                )
+            ),
+            connectable,
+            index_col="span_id",
+        )
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py:169]: addressed in a8261dbb9301253625fa713991c2ed7831d12f8a

File: src/phoenix/server/api/routers/v1/evaluations.py [hunk 3]
```diff
@@ -105,25 +116,71 @@ async def get_evaluations(request: Request) -> Response:
...
+
+    async with db() as session:
+        connection = await session.connection()
+        evaluation_dataframes: Dict[EvaluationName, DataFrame] = {}
+        for evaluation_name in span_evaluation_names:
+            evaluation_dataframe = await connection.run_sync(
+                read_sql_span_evaluations_into_dataframe, evaluation_name
+            )
+            if not evaluation_dataframe.empty:
+                evaluation_dataframes[evaluation_name] = evaluation_dataframe
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py]: I have not tried this, but my understanding is that SQLite won't let us share the connection in this way. Is that right @RogerHYang?

File: src/phoenix/session/client.py [hunk 4]
```diff
@@ -138,7 +138,7 @@ def get_evaluations(
             return session.get_evaluations(project_name=project_name)
         response = self._session.get(
             urljoin(self._base_url, "/v1/evaluations"),
-            params={"project_name": project_name},
+            params={"project-name": project_name},
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/session/client.py]: addressed in 3fa3d5bb6ad2c4b1263b1b81d89ede32ddc9f8fd

File: src/phoenix/server/api/routers/v1/evaluations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py]: Created ticket #2901

File: src/phoenix/server/api/routers/v1/evaluations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (RogerHYang) [reply src/phoenix/server/api/routers/v1/evaluations.py]: yes, we can only run this serially with a single session object

[the remark this answers is not in the document]
Comment (mikeldking) [reply src/phoenix/server/api/routers/v1/evaluations.py]: Gotcha

[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py]: The other option here is we can do a single query to pull all of a project's span annotations and do a groupby on the dataframe in memory.

[the remark this answers is not in the document]
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/v1/evaluations.py]: addressed in cfa3306ca20ee0d67493b4fdd70202b1b78e3ab2

File: src/phoenix/server/api/routers/v1/evaluations.py [hunk 5]
```diff
@@ -158,3 +176,31 @@ async def _add_evaluations(
...
+        .join(models.Span)
+        .join(models.Trace)
+        .join(models.Project)
+        .where(
+            and_(
+                models.Project.name == project_name,
+                models.SpanAnnotation.annotator_kind == "LLM",
+            )
+        ),
+        connectable,
```
Comment (RogerHYang) [inline src/phoenix/server/api/routers/v1/evaluations.py:204]: just to help pandas out a little bit
```suggestion
        )
        .order_by(models.SpanAnnotation.name),
        connectable,
```

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/api/routers/v1/evaluations.py (hunk 5 above)
Comment (RogerHYang) [reply src/phoenix/server/api/routers/v1/evaluations.py:204]: turns out this is unnecessary

**Pull request merged**